File: Java/versandetiketten/src/IO/EtikettDruckerStub.java
package IO;

import java.util.Iterator;
import java.util.List;

// ------------------------------------------------------------
// IO-Operation: Stub eines Etikettdruckers für die Demo
//
// Hier würde üblicherweise ein Interface
// zu einem echten Drucker bedient.
//-------------------------------------------------------------
public class EtikettDruckerStub implements IEtikettDrucker {

	@Override
	public void druckeEtikett(List<String> etikettZeilen) {
		Iterator<String> iter = etikettZeilen.iterator();
		
		while (iter.hasNext()) {
			String anschriftZeile = iter.next();
			System.out.println(anschriftZeile);
		}
		
		System.out.println("------------------------------------------");
	}
}

File: Csharp/VersandEtiketten/IO_Operationen/EtikettDruckerStub.cs
namespace IO_Operationen
{
    /// <summary>
    /// IO-Operation: Stub eines Etikettdruckers für die Demo
    //
    /// Hier würde üblicherweise ein Interface
    /// zu einem echten Drucker bedient.
    /// </summary>
    public class EtikettDruckerStub : IEtikettDrucker
    {
        public void DruckeEtikett(List<String?> etikettZeilen)
        {
            foreach (String? anschriftZeile in etikettZeilen) 
            { 
                Console.WriteLine(anschriftZeile);
            }

            Console.WriteLine("------------------------------------------");
        }
    }
}